### PR TITLE
Add simple validation against samples

### PR DESCRIPTION
**Pull request by willosborne**
Addresses issue #35 - simple rule on all samples. 
Currently doesn't do pattern files as the rules don't understand the json metaschema syntax yet.

### CODE REVIEW

**Comment by willosborne** [comment]
Note - branch is passing even though validation fails because it doesn't seem to have permission. 
https://github.com/stoplightio/spectral-action/issues/624 Seems others have had this problem. 

Concern that this may just not work for PRs from forks: https://github.com/italia/developers-italia-api/issues/67

File: .github/workflows/spectral-validation.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Run Spectral Validation on all samples
+
+on:
+  - pull_request
```
Comment (matthewgardner) [inline .github/workflows/spectral-validation.yml:4]:
```suggestion
  - pull_request
   - push:
      branches:
        - main
```

Comment (matthewgardner) [reply .github/workflows/spectral-validation.yml:4]: Should we also not run this again main when merged? (perhaps over kill but also nice to know the main branch is correct:) )

Comment (willosborne) [reply .github/workflows/spectral-validation.yml:4]: Yeah I think so.
We should run this 
 - when a PR is raised
 - when any new commits are pushed to said PR
 - when a PR is merged

**Review by rocketstack-matt — APPROVED** [review]

**Pull request merged**